### PR TITLE
fix boolean to Boolean Issue #189

### PR DESCRIPTION
**Pull request by abdelrahmanmohamed**
the default value of boolean is false, that means that the serializer will always serialize it to false in every request example:
```
Ticket x=new Ticket();
x.setId(ticket.getId());
```
json will be
```
{
  "tickets" : [ {
    "id" : 409671,
    "has_incidents" : false
  } ]
}
```
because of `private boolean hasIncidents;` need to be change to `private Boolean hasIncidents;` also the setter and the getters

### CODE REVIEW

**Comment by aheritier** [comment]
@abdelrahmanmohamed you mean that by replacing boolean -> Boolean by default they are null thus not serialised any more ?
I imagine in that case using primitive types for long/int/... (if any) then we should have the same issue...
Our automated tests (which are a bit light) are ok with your change but this PR changes the API and it might break various consumers who weren't protected boolean against null checks.
wdyt @stephenc ?

**Comment by abdelrahmanmohamed** [comment]
Yes, you will have the same issue if you are using primitive types int,long,double,byte ....etc and it will be serialised to zero, and that is the normal and expected behaviour, if you have any model that use primitive types it should be fixed by using wrapper classes unless the attribute have default business value

**Comment by stephenc** [comment]
We can make breaking changes as this is a 0.x.y version... though it is better to stack them all up

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,9 @@
     <developer>
       <name>Yoann Dubreuil</name>
     </developer>
+    <developer>
```
Comment (johnou) [inline pom.xml]: perhaps you mean contributor? :)

**Comment by johnou** [comment]
@abdelrahmanmohamed can you give me some actual use cases of where these changes help?

**Comment by johnou** [comment]
I'm not convinced, if you have a good few use cases for this pull request please reopen and comment.

**Comment by abdelrahmanmohamed** [comment]
@johnou lets say you have this class
```
class Ticket{
     int TicketId;
     boolean isOpened;
     String email;
}
```
the value for isOpened is true in the server
and you want to update the email using this model so you want to only send the ticketId and the email fields but when the json serializer serialize  the model it will found a default value for the isOpened = false so it will serialize it and that will cause the isOpened to be updated  to false

**Comment by johnou** [comment]
The isOpened field doesn't exist in ticket, I would rather do this on a case by case basis instead of converting all booleans to Booleans, probably even possible without a major release eg.

```java
public class Ticket extends Request implements SearchResultEntity {

    @JsonProperty("has_incidents")
    private Boolean hasIncidents;

    public boolean isHasIncidents() {
        return hasIncidents != null ? hasIncidents : false;
    }

    public void setHasIncidents(boolean hasIncidents) {
        this.hasIncidents = hasIncidents;
    }
}
```

**Pull request closed without merging**